File: fixtures/scale/dart/medium/lib/io/output_formatter.dart
import '../config/settings.dart';
import '../util/number_format.dart';

class OutputFormatter {
  final Settings settings;

  OutputFormatter(this.settings);

  String format(double value) {
    if (value.isNaN) return 'undefined';
    if (value.isInfinite) return value > 0 ? 'infinity' : '-infinity';
    final text = value.toStringAsFixed(settings.precision);
    return trimTrailingZeros(text);
  }
}

File: fixtures/scale/java/medium/src/calc/io/OutputFormatter.java
package calc.io;

import calc.config.Settings;

import static calc.util.NumberFormatUtil.trimTrailingZeros;

public class OutputFormatter {
    private final Settings settings;

    public OutputFormatter(Settings settings) {
        this.settings = settings;
    }

    public String format(double value) {
        if (Double.isNaN(value)) {
            return "undefined";
        }
        if (Double.isInfinite(value)) {
            return value > 0 ? "infinity" : "-infinity";
        }
        String text = String.format("%." + settings.getPrecision() + "f", value);
        return trimTrailingZeros(text);
    }
}
